vizro-ai/src/vizro_ai/plot/_response_models.py

```diff
@@ -36,6 +36,17 @@ def _format_and_lint(code_string: str) -> str:
     return formatted
 
 
+def _strip_markdown(code_string: str) -> str:
+    """Strip markdown code block from the code string."""
+    if code_string.startswith("```python\n"):
+        code_string = code_string[len("```python\n") :]
+    elif code_string.startswith("```\n"):
+        code_string = code_string[len("```\n") :]
+    if code_string.endswith("```"):
+        code_string = code_string[:-3]
+    return code_string.strip()
```

Comment:
```suggestion
def _strip_markdown(code_string: str) -> str:
    """Strip markdown code block from the code string."""
    if code_string.startswith("```python\n"):
        code_string = code_string[len("```python\n") :]
    elif code_string.startswith("```py\n"):
        code_string = code_string[len("```py\n") :]    
    elif code_string.startswith("```\n"):
        code_string = code_string[len("```\n") :]

    if code_string.endswith("```"):
        code_string = code_string[:len("```")]

    return code_string.strip()
```

Reply:
@petar-qb There is a bug in your suggestion ;)

Reply:
Good catch. There's missing `-` in this line ` code_string = code_string[:len("```")]` -> ` code_string = code_string[:-len("```")]`.  

I edited the suggestion.

Reply:
This could be refactored further.
```suggestion
def _strip_markdown(code_string: str) -> str:
    """Strip markdown code block from the code string."""    
    for strip_start_element in ["```python\n", "```py\n", "```\n"]:
        if code_string.startswith(strip_start_element):
            code_string = code_string[len(strip_start_element) :]
    
    for strip_end_element in ["```"]:
        if code_string.endswith(strip_end_element):
        code_string = code_string[:-len(strip_end_element)]

    return code_string.strip()
```